mne/time_frequency/spectrum.py

```diff
@@ -1179,6 +1179,56 @@ def __getitem__(self, item):
         return BaseRaw._getitem(self, item, return_times=False)
 
 
+@fill_doc
+class SpectrumArray(Spectrum):
+    """Spectrum object from numpy array.
+
+    Parameters
+    ----------
+    data : array, shape (n_channels, n_freqs)
+        The channels' power spectral density for each epoch.
+    %(info_not_none)s
+    %(freqs_tfr)s
+    %(method_psd)s
+    %(verbose)s
+
+    See Also
+    --------
+    mne.create_info
+    mne.EpochsArray
+    mne.EvokedArray
+    mne.io.RawArray
+    EpochsSpectrumArray
+
+    Notes
+    -----
+        .. versionadded:: 1.5.0
+    """
+
+    @verbose
+    def __init__(
+        self,
+        data,
+        info,
+        freqs,
+        method=None,
+        *,
+        verbose=None,
+    ):
+        self.__setstate__(
+            dict(
+                method=method,
+                data=data,
+                sfreq=info["sfreq"],
+                dims=("channel", "freq"),
+                freqs=freqs,
+                inst_type_str="Raw",
+                data_type="Average Power Spectrum",
```

Comment:
If the data comes from an array we don't know if it was computed on Raw or averaged (Evoked) data. Also we don't know if it's power/amplitude, real/complex, etc.

```suggestion
                inst_type_str="Array",
                data_type="Unknown",
```

This will probably break some tests (assuming we're testing thoroughly) so hopefully those test failures can guide what needs to change elsewhere

Reply:
So real/complex can be inferred from the data type. We ask specifically for power in the constructor so maybe we should require that let the `amplitude` kwarg in `Spectrum.plot` handle that? That'd be my 2 cents but not highly opinionated, whatever is simplest